main.py

```
@@ -15,6 +20,24 @@ class Plugin:
+            if self.settings_file_path.exists():
+                return json.loads(self.settings_file_path.read_text())
+        except Exception as e:
+            decky.logger.error(f"Error reading settings file: {e}")
+        return {"auto_lock_on_startup": False}
+
+    def _save_settings(self, data: PluginSettings) -> None:
+        """Persist plugin settings to disk."""
+        try:
+            self.settings_file_path.parent.mkdir(parents=True, exist_ok=True)
+            _ = self.settings_file_path.write_text(json.dumps(dict(data)))
+        except Exception as e:
```

Ruff (0.15.4)

[warning] Do not catch blind exception: `Exception`

(BLE001)